dusty/src/main.rs

```diff
@@ -66,15 +69,23 @@ fn main() {
 }
 
 // Extract probe records from the given file, if possible.
-pub(crate) fn probe_records<P: AsRef<Path>>(file: P) -> Result<Section, UsdtError> {
-    let data = fs::read(file)?;
-    let section = locate_probe_section(&data).ok_or(UsdtError::InvalidFile)?;
+pub(crate) fn probe_records<P: AsRef<Path>>(path: P) -> Result<Section, UsdtError> {
+    let file = OpenOptions::new()
+        .read(true)
+        .write(true)
```

Comment:
this is scary (ish?) we're opening the on-disk file RW so that we can treat it as `mut`... but we pinky-swear that we're not going to modify the file! because `register` is `false`! for realsies!

But if we're going to use `unsafe` anyways, why not just map it RO and lie about it being `mut`?

Reply:
Talked through this offline. We can use a private, COW mapping, and then open the file RO. That should never write through to disk, even if we hold this all incorrectly.